javaparser-core/src/main/java/com/github/javaparser/ast/CompilationUnit.java

```diff
@@ -465,6 +469,26 @@ public AnnotationDeclaration addAnnotationDeclaration(String name, Modifier... m
         return getTypes().stream().filter(type -> type.getNameAsString().equals(enumName) && type instanceof EnumDeclaration).findFirst().map(t -> (EnumDeclaration) t);
     }
 
+    /**
+     * @return the name that the primary type in this file should have, according to the filename in {@link Storage#getFileName()}.
+     * Empty if no file information is present (when this compilation unit wasn't parsed from a file.)
+     */
+    public Optional<String> getPrimaryTypeName() {
+        return getStorage()
+                .map(Storage::getFileName)
+                .map(filename -> filename.replace(".java", ""))
+                .map(filename -> filename.replace(".jav", ""));
```

Comment:
Yeah, I don't know...

Reply:
Yeah, why?

Reply:
Maybe on DOS systems they are limited to extensions of 3 characters